libs/optimizer/src/lib/utils/createSegments.ts

```diff
@@ -0,0 +1,51 @@
+import { LatLonAltTime, ScoringTrack } from '../optimizer';
+
+
+/**
+ * Create segments between 2 points.
+ * Added points are computed by a linear interpolation
+ * @param from start point of the segment
+ * @param to end point
+ * @param minTimeSec time in seconds since 1970-01-01T00:00:00.000 when the track starts
+ * @param nbSegments number of segments created. If nbSegments <= 1 the result contains one segment ('from' -> 'to')
+ * @param distributionFactor if nbSegments > 1, this factor will narrow the distance between generated points and the start
+ *                           point of the segment. if close to zero, points will be close to the start point.
+ *                           if equals 1, points will be equally distributed along the segment.
+ * @return a ScoringTrack of nbSegments
+ */
+export function createSegments(
+  from: LatLonAltTime,
+  to: LatLonAltTime,
+  minTimeSec: number,
+  nbSegments: number,
+  distributionFactor: number = 1,
+): ScoringTrack {
+  const result: ScoringTrack = { points: [], startTimeSec: minTimeSec };
+
+  appendToResult(from);
+
+  if (nbSegments > 1) {
+    appendIntermediatePoints();
+  }
+  appendToResult(to);
+  return result;
+
+  function appendToResult(p: LatLonAltTime) {
```

Comment:
nit: inlining those functions would be more readable

Reply:
Ok. Changed